protocol.go

```diff
@@ -202,6 +202,11 @@ func (session *session) handleMAIL(cmd command) {
 		return
 	}
 
+	if session.server.Authenticator != nil && session.peer.Username == "" {
+		session.reply(530, "Authentication Required.")
+		return
+	}
```

Comment:
Do you mind adding a test for this?

Reply:
Sure, I will try to come up with a test.